articles/quickstart/spa/angular-beta/02-calling-an-api.md

```diff
@@ -0,0 +1,284 @@
+      }
+    }
+  },
+...
+```
+
+With this in place, the frontend application can make a request to `/api/external` and it will be correctly proxied through to the backend API at `http://localhost:3001/api/external`.
+
+## Modify the AuthService Class
+
+Modify the `src/app/auth.service.ts` file so that the `audience` value is passed through to the Auth0 client:
+
+```js
+// src/app/auth.service.ts
+
+this.auth0Client = await createAuth0Client({
+  domain: "${account.tenant}",
+  client_id: "${account.clientId}"
+  audience: "${apiIdentifier}" // NEW - add in the audience value
+});
+```
+
+## Calling the API
+
+Now create a new component that can be shown on the screen to allow the user to call the backend API:
+
+```bash
+ng generate component external-api
+```
+
+Open `src/app/external-api/external-api.component.html` and replace its contents with the following:
+
+```html
+<!-- src/app/external-api/external-api.component.html -->
+
+<div>
+  <button (click)="pingApi()">Ping API</button>
+
+  <div>
+    <div [ngClass]="{ show: hasResponse }">
+      <h6 class="muted">Result</h6>
+      <pre>
+        <code>{{ responseJson }}</code>
+      </pre>
+    </div>
+  </div>
+</div>
+```
+
+The UI for this component is a simple button that invokes a call to the API. The results of the call are then displayed in the `code` element.
+
+Next, create a new service that will be responsible for making API calls:
+
+```bash
+ng generate service api
+```
+
+Then open `src/app/api.service.ts` and replace its contents with the following:
+
+```ts
+// src/app/api.service.ts
+
+import { Injectable } from '@angular/core';
+import { AuthService } from './auth.service';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ApiService {
+  constructor(
+    private authService: AuthService,
+    private httpClient: HttpClient
+  ) {}
+
+  async ping(): Promise<any> {
+    const client = await this.authService.getAuth0Client();
+    const token = await client.getTokenSilently();
+
+    return this.httpClient
+      .get('/api/external', {
+        headers: {
+          Authorization: `Bearer <%= "${token}" %>`
+        }
+      })
+      .toPromise();
+  }
+}
+```
+
+All the endpoints that the API exposes can be added here. Right now, there is just one: the `ping` endpoint. The `ping()` method above retrieves the current access token from the `AuthService` instance and then uses `HttpClient` to make a GET request to the backend API, passing the access token inside the [Authorization header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization).
+
+For `HttpClient` to work properly, the `HttpClientModule` should also be included in the application. To do this, open `app.module.ts` and import the module. Then, add the module to the list of imports as in the declaration of your `AppModule`.
+
+With these changes, `app.module.ts` should look something like the following:
```

Comment:
> ... should look something like the following

If that's the entire file, then it should look **exactly** like the following. If there is only a simple line change, is it necessary to show the entire file?

Reply:
@joshcanhelp I've reduced the sample down. You're right, I'm not sure it's necessary to show the whole file.